pkg/api/v1/defaults.go

```diff
@@ -389,3 +389,30 @@ func SetDefaults_HostPathVolumeSource(obj *v1.HostPathVolumeSource) {
 		obj.Type = &typeVol
 	}
 }
+
+func SetDefaults_SecurityContext(sc *v1.SecurityContext) {
+	pt := true
+	if sc == nil {
+		sc.AllowPrivilegeEscalation = &pt
+		return
+	}
+
+	// handle the case for AllowPrivilegeEscalation when the container is privileged
+	if sc.Privileged != nil && *sc.Privileged {
+		sc.AllowPrivilegeEscalation = &pt
```

Comment:
Hmm, I'm not sure it's acceptable for defaulting code to change values that have been set. Maybe it would be better for this to go in validation (i.e. setting `AllowPrivilegeEscalataion == false &&  Privileged == true` is an error)

Reply:
can do

Reply:
updated in a differnt commit, can squash before merge, added tests too